public/app/core/utils/UserProvider.tsx

```diff
@@ -101,6 +109,62 @@ export class UserProvider extends PureComponent<Props, State> {
     this.setState({ orgs, loadingStates: { ...this.state.loadingStates, loadOrgs: false } });
   };
 
+  loadSessions = async () => {
+    this.setState({
+      loadingStates: { ...this.state.loadingStates, loadSessions: true },
+    });
+
+    await getBackendSrv()
+      .get('/api/user/auth-tokens')
+      .then((sessions: UserSession[]) => {
+        sessions.reverse();
+
+        const found = sessions.findIndex((session: UserSession) => {
+          return session.isActive;
+        });
+
+        if (found) {
+          const now = sessions[found];
+          sessions.splice(found, found);
+          sessions.unshift(now);
+        }
+
+        sessions = sessions.map((session: UserSession) => {
+          return {
+            id: session.id,
+            isActive: session.isActive,
+            seenAt: dateTimeFormatTimeAgo(session.seenAt),
+            createdAt: dateTimeFormat(session.createdAt, { format: 'MMMM DD, YYYY' }),
+            clientIp: session.clientIp,
+            browser: session.browser,
+            browserVersion: session.browserVersion,
+            os: session.os,
+            osVersion: session.osVersion,
+            device: session.device,
+          };
+        });
+
+        this.setState({ sessions, loadingStates: { ...this.state.loadingStates, loadSessions: false } });
+      });
+  };
+
+  revokeUserSession = async (tokenId: number) => {
```

Comment:
@peterholmberg at the time I started the conversion of the user profile, it was decided that redux was not necessary for this as the states were not used outside of the page and @dprokop had me create the UserProvider as the place where all of this would be housed.

Reply:
Ok 👌